src/searchbar.cpp

```diff
@@ -190,8 +191,20 @@ void SearchBarLineEdit::fetchMoreSuggestions()
     fetchSuggestions(&SearchBarLineEdit::onAdditionalSuggestions);
 }
 
+void SearchBarLineEdit::setNoMoreSuggestion(int fetchedSize)
+{
+    int trueFetchSize = fetchedSize - (m_suggestionModel.hasFullTextSearchSuggestion() ? 1 : 0);
+    m_noMoreSuggestion = trueFetchSize < SuggestionListWorker::getFetchSize();
+}
```

Comment:
Such a method is not fool proof. Its implementation assumes that it is called after the fetched suggestions have been appended to `m_suggestionModel`. Always try to avoid such tacit dependencies between members of your API. I see no justification for this member function. Just inline it in its sole point of usage and compute the status without any reference to `m_suggestionModel` (use only the incoming list of suggestions).

Reply:
Done